lib/iris/tests/unit/plot/test_2d_coords.py

```diff
@@ -135,32 +114,120 @@ def test_map_common_2d(self):
         result = iplt._map_common('pcolor', None, self.mode, cube, plot_defn)
         self.assertTrue(result)
 
-#    def test_discontiguous_masked(self):
-#        # Test that a contiguity check will raise a warning (not an error) for
-#        # discontiguous bounds but appropriately masked data.
-#        cube = self.latlon_2d
-#        coord = cube.coord('longitude')
-#        msg = 'The bounds of the longitude coordinate are not contiguous.  ' \
-#              'However, data is masked where the discontiguity occurs so ' \
-#              'plotting anyway.'
-#        with self.assertWarnsRegexp(msg):
-#            iplt._check_contiguity_and_bounds(coord, cube.data)
+    def test_draw_2d_from_bounds(self):
+        # Test this function will not raise an error even with our most
+        # awkward but supported cube.
+        cube = self.latlon_2d
+        result = iplt._draw_2d_from_bounds('pcolormesh', cube)
+        self.assertTrue(result)
```

Comment:
`draw_2d_from_bounds` is too complicated to unit test. I suggest you just remove this test.
I am in the process of adding an integration test that will test `draw_2d_from_bounds` (which will at the same time check that it is not raising an error)

Reply:
Yeah, this was meant as a placeholder really, I think it was one of the first tests I put in, just to help me think.  I'll take it out.